src/AddressRegistry.Projections.Elastic/AddressList/AddressListDocument.cs

```diff
@@ -0,0 +1,212 @@
+                        break;
+                    case PostalLanguage.French:
+                        names.Add(new Name(postalName.PostalName!, Language.fr));
+                        break;
+                    case PostalLanguage.German:
+                        names.Add(new Name(postalName.PostalName!, Language.de));
+                        break;
+                    case PostalLanguage.English:
+                        names.Add(new Name(postalName.PostalName!, Language.en));
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            return new PostalInfo(postalInfoLatestItem.PostalCode, names);
+        }
+    }
+
+    public sealed class StreetName
+    {
+        public int StreetNamePersistentLocalId { get; set; }
+        public Name[] Names { get; set; }
+
+        public Name[] HomonymAdditions { get; set; }
+
+        public StreetName()
+        { }
+
+        public StreetName(int streetNamePersistentLocalId, IEnumerable<Name> names, IEnumerable<Name> homonymAdditions)
+        {
+            StreetNamePersistentLocalId = streetNamePersistentLocalId;
+            Names = names.ToArray();
+            HomonymAdditions = homonymAdditions.ToArray();
+        }
+
+        public static StreetName FromStreetNameLatestItem(StreetNameLatestItem streetName)
+        {
+            return new StreetName(
+                streetName.PersistentLocalId,
+                new[]
+                    {
+                        new Name(streetName.NameDutch, Language.nl),
+                        new Name(streetName.NameFrench, Language.fr),
+                        new Name(streetName.NameGerman, Language.de),
+                        new Name(streetName.NameEnglish, Language.en)
+                    }
+                    .Where(x => !string.IsNullOrEmpty(x.Spelling)),
+                new[]
+                    {
+                        new Name(streetName.HomonymAdditionDutch, Language.nl),
+                        new Name(streetName.HomonymAdditionFrench, Language.fr),
+                        new Name(streetName.HomonymAdditionGerman, Language.de),
+                        new Name(streetName.HomonymAdditionEnglish, Language.en)
+                    }
+                    .Where(x => !string.IsNullOrEmpty(x.Spelling))
+            );
+        }
+    }
+
+    public sealed class AddressPosition
+    {
+        public string GeometryAsWkt { get; set; }
+        public string GeometryAsWgs84 { get; set; }
+        public GeometryMethod GeometryMethod { get; set; }
+        public GeometrySpecification GeometrySpecification { get; set; }
+
+        public AddressPosition()
+        { }
+
+        public AddressPosition(
+            Point point,
+            GeometryMethod geometryMethod,
+            GeometrySpecification geometrySpecification)
+        {
+            GeometryAsWkt = point.AsText();
```

Comment:
Kunnen we deze logica naar iets gezamenlijk verhuizen voor list en search documents?